src/rnnlm/sample_a_word.cc

```diff
@@ -0,0 +1,432 @@
+// sample_a_word.cc
+
+#include "sample_a_word.h"
+#include <iostream>
+#include <string>
+#include <iterator>
+#include <algorithm>
+#include <math.h>
+
+// Constructor for sampling the next word
+NgramModel::NgramModel(char* arpa_file, char* histories_file) {
+  vocab_size_ = 0;
+  ReadARPAModel(arpa_file);
+  ReadHistories(histories_file);
+}
+
+// Read language model from a ARPA-format file.
+void NgramModel::ReadARPAModel(char* file) {
+  std::ifstream data_input(file);
+  if (!data_input.is_open()) {
+    std::cerr << "error opening '" << file
+              << "' for reading\n";
+    exit(1);
+  }
+  std::string line;
+  int32 order;
+  int32 order_current = 0;
+  int32 word;
+  int32 iter = 0;
+  int32 while_iter = 0;
+  std::pair <float, float> probs_pair;
```

Comment:
Hmm, my guess is because Guoguo's code interacts with OpenFST, and it seems OpenFST doesn't have full support for double's?
@danpovey I was wondering what you think?